### ISSUE TITLE
Tessera leaks memory when converting vector tiles into raster tiles

### ISSUE THREAD

**Issue by hannesj** [mojodna/tessera#19]
I'm currently trying to serve raster tiles based off MapBox-formatted vector tiles, but after a while tessera crashes with the following:

    GET /hsl-map/13/4661/2372.png 200 1362.108 ms - 2521
    GET /hsl-map/13/4660/2372.png 200 1119.578 ms - 2209
    GET /hsl-map/13/4666/2372.png 200 1261.209 ms - 5300
    GET /hsl-map/13/4665/2372.png 200 1324.336 ms - 7149
    FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - process out of memory
    Aborted (core dumped)

I'm running on 64-bit Debian Jessie with node 0.12. The TileJSON for the source can be found at http://tile3.kartat.kapsi.fi/mapbox/index.json and the style is https://github.com/HSLdevcom/openjourneyplanner-map. Am I missing some options for node to increase the memory limits, or might some component be leaking memory?

**Comment by mojodna** [comment]
This looks like the same issue as https://github.com/mojodna/tl/issues/3, which is fundamentally a `tilelive-cache` problem, tracked here: https://github.com/mojodna/tilelive-cache/issues/3

(I haven't had time to look into fixing it yet.)

The short-term fix may be to remove `tilelive-cache` from https://github.com/mojodna/tessera/blob/master/server.js#L19-L22, similar to https://github.com/mojodna/tl/issues/3#issuecomment-74574620, changing

```javascript
      tilelive = require("tilelive-cache")(require("tilelive"), {
        size: process.env.CACHE_SIZE || opts.cacheSize,
        sources: process.env.SOURCE_CACHE_SIZE || opts.sourceCacheSize
      });
```

to

```javascript
      tilelive = require("tilelive");
```

That *may* slow things down slightly. but it shouldn't run out of memory.

**Comment by mojodna** [comment]
Fixed via tilelive-cache@0.4.3.  Reinstalling `tessera` will pick up the new version of the dependency.

**Comment by hannesj** [comment]
Thanks, seems to be working properly now.
